common/src/main/java/co/elastic/otel/common/SpanValue.java

```diff
@@ -271,9 +272,13 @@ private static Span unwrap(Object span) {
     if (span instanceof MutableSpan) {
       return unwrap(((MutableSpan) span).getOriginalSpan());
     }
+    if (CONTRIB_MUTABLE_SPAN_CLASS != null && CONTRIB_MUTABLE_SPAN_CLASS.isInstance(span)) {
```

Comment:
I'd suggest that we add a unit test for this logic:

It should be as easy as just adding a new test arguments [here](https://github.com/elastic/elastic-otel-java/blob/e61b4bd32b2a12457152a052613bde77da59f165/common/src/test/java/co/elastic/otel/common/SpanValueTest.java#L116):
```
Arguments.of(
            Named.of(
                "Contrib MutableSpan, sparse SpanValue",
                ValueAccess.create(
                    SpanValue.<String>createSparse(),
                    io.opentelemetry.contrib.stacktrace.internal.MutableSpan.makeMutable((ReadableSpan) newSpan())))));
```